aries_cloudagent/messaging/jsonld/routes.py

```diff
@@ -3,26 +3,44 @@
 from aiohttp import web
 from aiohttp_apispec import docs, request_schema, response_schema
 
-from marshmallow import fields
+from marshmallow import fields, Schema
 
 from ...admin.request_context import AdminRequestContext
-from ...wallet.base import BaseWallet
 
 from ..models.openapi import OpenAPISchema
-
+from ...config.base import InjectionError
+from ...wallet.error import WalletError
+from ...resolver.did_resolver import DIDResolver
+from ...resolver.base import ResolverError
+from .error import MissingVerificationMethodError, BaseJSONLDMessagingError
 from .credential import sign_credential, verify_credential
-from .error import (
-    BadJWSHeaderError,
-    DroppedAttributeError,
-    MissingVerificationMethodError,
-)
 
 
 class SignRequestSchema(OpenAPISchema):
     """Request schema for signing a jsonld doc."""
 
-    verkey = fields.Str(required=True, description="Verkey to use for signing")
-    doc = fields.Dict(required=True, description="JSON-LD document to sign")
+    verkey = fields.Str(required=True, description="verkey to use for signing")
+    doc = fields.Nested(
+        Schema.from_dict(
```

Comment:
all these years and I never knew about Schema.from_dict()!

Reply:
Schema.from_dict() is nice when it works. I find it convenient to use. Let me know if it goes against conventions.